tools/rv_histogram.c

```diff
@@ -0,0 +1,276 @@
+#include <errno.h>
+#include <getopt.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+
+#include "../src/decode.h"
+#include "../src/elf.h"
+
+#define ARRAYS_SIZE(arr) (sizeof(arr) / sizeof(arr[0]))
+
+/* 5 are fuse instructions */
+#define RISCV_INSN_ACTUAL_SIZE N_RISCV_INSN_LIST - 5
+
+static bool ascending_order = false;
+static const char *elf_prog = NULL;
+static const char *optstr = "a";
+
+typedef struct {
+    char insn_str[16];
+    size_t freq;
+} rv_insn_hist_t;
+
+static rv_insn_hist_t rv_insn_str_and_freq_tbl[] = {
+    {"nop", 0},       {"lui", 0},      {"auipc", 0},    {"jal", 0},
```

Comment:
Use [X-macro](https://en.wikipedia.org/wiki/X_macro) trick and [Stringizing](https://gcc.gnu.org/onlinedocs/cpp/Stringizing.html) to generate this table. Of course, you have to skip fuse instructions when manipulations.